src/main/java/gov/cabinetoffice/gap/adminbackend/security/JwtTokenFilter.java

```diff
@@ -0,0 +1,69 @@
+package gov.cabinetoffice.gap.adminbackend.security;
+
+import gov.cabinetoffice.gap.adminbackend.config.JwtTokenFilterConfig;
+import gov.cabinetoffice.gap.adminbackend.exceptions.ForbiddenException;
+import gov.cabinetoffice.gap.adminbackend.exceptions.UnauthorizedException;
+import gov.cabinetoffice.gap.adminbackend.models.AdminSession;
+import gov.cabinetoffice.gap.adminbackend.services.JwtService;
+import gov.cabinetoffice.gap.adminbackend.services.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.authentication.logout.SecurityContextLogoutHandler;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.springframework.util.ObjectUtils.isEmpty;
+
+/**
+ * This class cannot be a Spring bean, otherwise Spring will automatically apply it to all
+ * requests, regardless of whether they've been specifically ignored
+ */
+@RequiredArgsConstructor
+public class JwtTokenFilter extends OncePerRequestFilter {
+
+    private final JwtService jwtService;
+
+    private final UserService userService;
+
+    private final JwtTokenFilterConfig jwtTokenFilterConfig;
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain chain)
+            throws ServletException, IOException {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+
+        if (authentication == null || !authentication.isAuthenticated()) {
+            chain.doFilter(request, response);
+            return;
+        }
+
+        AdminSession adminSession = ((AdminSession) authentication.getPrincipal());
+        if (!jwtTokenFilterConfig.oneLoginEnabled) {
+            chain.doFilter(request, response);
```

Comment:
We are returning the same as line 46 here?

What happens if One Login is turned off?

Reply:
If onelogin is turned off we will hit chain.doFilter which will skip the verifyRoles call. Users only have roles in after OL migration so we want to let them through with any valid admin session. 

Have added that predicate up & into to the other if statement. tbh the check there for `!authentication.isAuthenticated() `may not even be required as the filter is only applied after the adminsession is verified